apps/backend/src/auth/auth.service.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  BadRequestException,
+  ConflictException,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { bcrypt } from 'bcryptjs';
+import { ParishService } from '../parish/parish.service';
+import { LoginDataDto, ParishDataDto } from './dto/login.dto';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly JwtService: JwtService,
+    private readonly parishService: ParishService
+  ) {}
+
+  /**
+   * This function authenticate the parish user when login
+   * @param input
+   * @returns
+   */
+  async authenticateParish(input: LoginDataDto): Promise<ParishDataDto> {
+    const user = await this.validate(input, 'parish');
+
+    if (!user) {
+      throw new BadRequestException('Bad Request', {
+        cause: new Error(),
+        description: 'Wrong email or password.',
+      });
+    } else if (typeof user === 'string') {
+      throw new ConflictException(user, {
+        cause: new Error(),
+        description:
+          'Your account is already in use! Please logout before logging again.',
+      });
+    }
+
+    return this.signIn(user);
+  }
+  /**
+   * This function authenticate administrators user when login
+   * @param input
+   * @returns
+   */
+  async authenticateAdmin(input: LoginDataDto): Promise<ParishDataDto> {
+    const user = await this.validate(input, 'admin');
+
+    if (!user) {
+      throw new BadRequestException('Bad Request', {
+        cause: new Error(),
+        description: 'Wrong email or password.',
+      });
+    } else if (typeof user === 'string') {
+      throw new ConflictException(user, {
+        cause: new Error(),
+        description:
+          'Your account is already in use! Please logout before logging again.',
+      });
+    }
+
+    return this.signIn(user);
+  }
+
+  /**
+   * This function validate weither the user exists in db or not.
+   * Alo check weither all user's credentials are valid or not.
+   * @param input
+   * @returns
+   */
+  async validate(
+    input: LoginDataDto,
+    role: string
+  ): Promise<ParishDataDto | null | string> {
+    const { email, password } = input;
+    const user = await this.parishService.findOneByMail(email);
+
+    if (!user) {
+      return null;
+    } else if (user.token) {
+      const verifyToken = await this.tokenCheckValidity(user.token, user.id);
+      if (typeof verifyToken === 'string') {
+        return verifyToken;
+      }
+    }
+
+    try {
+      const validatePassword = bcrypt.compare(password, user.password);
+      if (validatePassword) {
+        return {
+          id: user.id,
+          email: user.email,
+          city: user.city,
+          diocese: user.diocese,
+          leadManager: user.leadManager,
+          name: user.name,
+          phone: user.phone,
+          region: user.region,
+          createdAt: user.createdAt,
+        };
+      } else return null;
```

Comment:
Instead return the complete removing the password such that your return type changes to `User | null`, then move this logic else where.